org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,375 @@
+                    return providerSchemaToPlatformSchema(response, expand);
+                });
+        }
+    }
+
+    getDevicesAmbientTemperature(di) {
+        return getDeviceResource(this, di, 'ambientTemperature');
+    }
+
+    getDevicesTargetTemperature(di) {
+        return getDeviceResource(this, di, 'targetTemperature');
+    }
+
+    postDevicesTargetTemperature(di, payload) {
+        return postDeviceResource(di, 'targetTemperature', payload);
+    }
+
+    getDevicesHumidity(di) {
+        return getDeviceResource(this, di, 'humidity');
+    }
+
+    getDevicesTargetTemperatureHigh(di) {
+        return getDeviceResource(this, di, 'targetTemperatureHigh');
+    }
+
+    postDevicesTargetTemperatureHigh(di, payload) {
+        return postDeviceResource(di, 'targetTemperatureHigh', payload);
+    }
+
+    getDevicesTargetTemperatureLow(di) {
+        return getDeviceResource(this, di, 'targetTemperatureLow');
+    }
+
+    postDevicesTargetTemperatureLow(di, payload) {
+        return postDeviceResource(di, 'targetTemperatureLow', payload);
+    }
+
+    getDevicesAwayMode(di) {
+        return getDeviceResource(this, di, 'awayMode');
+    }
+
+    postDevicesAwayMode(di, payload) {
+        return postDeviceResource(di, 'awayMode', payload);
+    }
+
+    getDevicesAwayTemperatureHigh(di) {
+        return getDeviceResource(this, di, 'awayTemperatureHigh');
+    }
+
+    postDevicesAwayTemperatureHigh(di, payload) {
+        return postDeviceResource(di, 'awayTemperatureHigh', payload);
+    }
+
+    getDevicesAwayTemperatureLow(di) {
+        return getDeviceResource(this, di, 'awayTemperatureLow');
+    }
+
+    postDevicesAwayTemperatureLow(di, payload) {
+        return postDeviceResource(di, 'awayTemperatureLow', payload);
+    }
+
+    getDevicesEcoMode(di) {
+        return getDeviceResource(this, di, 'ecoMode');
+    }
+
+    getDevicesHvacMode(di) {
+        return getDeviceResource(this, di, 'hvacMode');
+    }
+
+    postDevicesHvacMode(di, payload) {
+        return postDeviceResource(di, 'hvacMode', payload);
+    }
+
+    getDevicesHeatingFuelSource(di) {
+        return getDeviceResource(this, di, 'heatingFuelSource');
+    }
+
+    getDevicesHasFan(di) {
+        return getDeviceResource(this, di, 'hasFan');
+    }
+
+    getDevicesFanActive(di) {
+        return getDeviceResource(this, di, 'fanActive');
+    }
+
+    getDevicesFanTimerActive(di) {
+        return getDeviceResource(this, di, 'fanTimerActive');
+    }
+
+    getDevicesFanTimerTimeout(di) {
+        return getDeviceResource(this, di, 'fanTimerTimeout');
+    }
+
+    postDevicesFanTimerTimeout(di, payload) {
+        return postDeviceResource(di, 'fanTimerTimeout', payload);
+    }
+
+    getDevicesFanMode(di) {
+        return getDeviceResource(this, di, 'fanMode');
+    }
+
+    postDevicesFanMode(di, payload) {
+        return postDeviceResource(di, 'fanMode', payload);
+    }
+
+    postSubscribe(callbackUrl, verificationRequest) {
```

Comment:
In the current design, the callback URL is set in the SmartApp.

---
In reply to: [90385111](https://github.com/openT2T/translators/pull/134#discussion_r90385111) [](ancestors = 90385111)